tools/auth/hubspot.go

```diff
@@ -0,0 +1,78 @@
+package auth
+
+import (
+	"encoding/json"
+	"errors"
+	"strings"
+
+	"golang.org/x/oauth2"
+)
+
+var _ Provider = (*Hubspot)(nil)
+
+// NameHubspot is the unique name of the Hubspot provider.
+const NameHubspot string = "hubspot"
+
+// Hubspot allows authentication via Hubspot OAuth2.
+type Hubspot struct {
+	*baseProvider
+}
+
+// NewHubspotProvider creates a new Hubspot provider instance with some defaults.
+func NewHubspotProvider() *Hubspot {
+	return &Hubspot{&baseProvider{
+		scopes: []string{
+			"crm.objects.owners.read",
```

Comment:
Maybe change to `"oauth"`?
_(https://developers.hubspot.com/docs/api/working-with-oauth#scopes)_